modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserTestCase.java

```diff
@@ -31,19 +54,45 @@
 import org.mule.test.heisenberg.extension.model.KnockeableDoor;
 import org.mule.test.vegan.extension.VeganCookBook;
 
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 
+import io.qameta.allure.Description;
+import io.qameta.allure.Issue;
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
 public class JavaExtensionModelParserTestCase {
 
+  private static final String COMPILATION_MODE = "COMPILATION_MODE";
+  private static Map<String, ExtensionModel> EXTENSION_MODELS = new HashMap<>();
+  private static final ExtensionModelLoader JAVA_LOADER = new DefaultJavaExtensionModelLoader();
+
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
 
+  @Test
+  @Issue("W-12622240")
+  @Description("Verify that ExtensionModel for an Extension with java data types such as LocalDateTime, loads the correct MetadataType i.e. DateTimeType and works in Java 17")
+  public void getParameterizedWithJavaFieldsExtensionUsingSdkApi() {
+    ExtensionModel extensionModel = loadExtension(ParameterizedWithJavaTypeExtension.class, JAVA_LOADER, null);
+    ConfigurationModel configModel = extensionModel.getConfigurationModels().iterator().next();
+    ParameterModel parameterModel = configModel.getAllParameterModels().iterator().next();
+    DefaultObjectType objectType = (DefaultObjectType) parameterModel.getType();
+    assertThat(objectType.toString(),
+               is("org.mule.runtime.module.extension.internal.loader.parser.java.JavaExtensionModelParserTestCase.SimplePojoWithTime"));
+    Collection<ObjectFieldType> fields = objectType.getFields();
+    List<String> simplePojoWithTimeFields = fields.stream().map(f -> f.getValue().getClass().getName()).collect(toList());
+
+    assertThat(simplePojoWithTimeFields, hasItem(equalTo("org.mule.metadata.api.model.impl.DefaultDateTimeType")));
```

Comment:
`DefaultDateTimeType` is api, you can reference the class instead of having to use its name as a string

Reply:
don't rely on toString() for the assertions. A change there should not cause tests to break. Check fields by actually asserting on the correct interface. Also include the fields name to make sure you're asserting over the correct thing.